### ISSUE TITLE
false negative `reportOverlappingOverload` with constraints

### ISSUE THREAD

**Issue by KotlinIsland** [DetachHead/basedpyright#750]
```py
from typing import overload


class A: ...
class B(A): ...

@overload
def f1(x: A) -> None: ...

@overload
def f1(x: B) -> None: ...  # Overload 2 for "f1" will never be used because its parameters overlap overload 1 (reportOverlappingOverload)

def f1(x: A):
    print(x)

def f2[T: (A, B)](x: T):  # no error
    print(x)
```

**Comment by KotlinIsland** [comment]
erm, what the spruce, constraints don't work the same as overloads, and instead find the most specific match, fail
